### PR TITLE
Translate Protocol.md to Chinese

### PR DESCRIPTION
**Pull request by reymondtu**
Translate Protocol.md to Chinese

### CODE REVIEW

File: Protocol_zh.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	1		对消息类型0的回复.这个回复必须在所有请求的区块之后.
+			
+	2		请求某个时间段内的所有区块的汇总信息.起始时间在区块头的时间字段中，结束时间在区块头的金额字段中.
+
+	3		对消息类型2的回复.回复是一个区块,其中的后256个字节是由16个汇总结构组成.每个汇总将总的时间段划分成16等份.
+			每个汇总信息包括两个内容:
+			- the first 64-bit little-endian number is the sum of all
+			  64-bit numbers of all blocks of this sub-interval;
+			- the second 64-bit little-endian number is the sum of lengths 
+			  of all blocks from this sub-interval.
```
Comment (xrdavies) [inline Protocol_zh.md]: Please check this section which is still not translated.

File: Protocol_zh.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+## 1.名词解释
+
+**区快地址** - 区块hash截取后的base64编码.
+
+**金额** - XDAG的数量，以Cheato为单位，类似比特币的Santoshi.  
+
+**区块** - Dagger的基本结构, 每个区块定长512字节, 结构定义在第二部分介绍.
+
+**链** - 前一个区块引用后一个区块区块，按一定的顺序组成链.
```
Comment (xrdavies) [inline Protocol_zh.md]: Chain - a sequence of blocks each block of which is referenced by the previous block.
**链** - 前一个区块引用后一个区块区块，按一定的顺序组成链.

Maybe need few modification to be more precise. Linked and referenced are different.

File: Protocol_zh.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+**金额** - XDAG的数量，以Cheato为单位，类似比特币的Santoshi.  
+
+**区块** - Dagger的基本结构, 每个区块定长512字节, 结构定义在第二部分介绍.
+
+**链** - 前一个区块引用后一个区块区块，按一定的顺序组成链.  
+
+**Cheato** - 一个XDAG包括2^32个Cheato.  
+
+**Dagger创世块** - Unix格式：0x5A500000 ，也就是2018年1月6号 22:45:20 GMT.
```
Comment (xrdavies) [inline Protocol_zh.md]: Dagger era - time 0x5A500000 in Unix format, i.e. January 6, 2018, 22:45:20 GMT.

Dagger era means “Dagger元年”, not genesis block.

File: Protocol_zh.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+**Dagger创世块** - Unix格式：0x5A500000 ，也就是2018年1月6号 22:45:20 GMT.  
+
+**区块难度** - (2^128-1)/(hash_little / 2^160), hash_little = 小端字节序(hash(block)), hash_little大小为256 bits(32 bytes).
+
+**链难度** - 链上区块的难度之和.  
+
+**Hash** - 采用双sha256哈希算法, hash(block) = sha256(sha256(block)).
+
+**主区块** - 主区块链上的任何区块.
```
Comment (xrdavies) [inline Protocol_zh.md]: Need keep the consistency of translation. Let's call the main block as "主块" and the main chain as “主链” to make it different from the concept from blockchain.

File: Protocol_zh.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+	2		交易的输入. 连接到另外一个区块B.  
+			结构如下:
+			- 192-bit 区块B的hash截断.
+			- 64-bit 小端字节序, 区块被应用后的余额.
+
+
+	3		交易的输出. 连接到其他区块. 
+	        结构如下:
+			- 192-bit 区块B的hash截断.
+			- 64-bit 小端字节序, 区块被应用后的余额.
```
Comment (xrdavies) [inline Protocol_zh.md]: - 64-bit little-endian number, **the amount that is taken from the balance of block B** then the transaction is applied.

It should be amount, not balance in translation.

File: Protocol_zh.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+每一个区块都是一个交易. 一个交易个可能有多个输入,、输出、公钥、 输入和输出的签名, 还有手续费. 如果满足以下条件，则区块是有效的:  
+
+- 区块的产生时间不小于Dagger的创世时间;
+- 区块的输入、输出时间小于区块A的产生时间;
+- 区块的每个输入或者输出必须是合法的区块;
+- 区块的输入金额总和必须小于2^64;
+- 区块的输出金额加上手续费必须小于2^64;
+- 区块至少有一个输入，所有输入金额的总和大于等于所有输出加上手续费的总和, 否则所有输出的总和必须为零;
+- 区块A的每个输入B，在区块A中存在公钥K和输入或输出签名S, 并在块B中输出签名T，使得使用密钥K从块A获得签名S，并且从块中获得签名T B使用相同的密钥K（非正式描述：只有B块的所有者可以从中提取资金）.
```
Comment (xrdavies) [inline Protocol_zh.md]: It's correctly translated but hard to read. Need optimization.

File: Protocol_zh.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+## 4.加密与安全
+
+签名算法，使用openssl实现中的标准ECDSA签名.椭圆曲线是Secp256k1.私钥长度为32个字节，公钥的长度为32字节（椭圆曲线上的点的x坐标）加上（椭圆曲线上的点的y坐标）.签名是64个字节，由两个字段组成.这两个字段别是ECDSA签名算法中的r和s.
+
+私钥存在wallet.dat文件中,这个文件和xdag可执行程序在同一目录.每个私钥长度32个字节, 公钥不保存，因为公钥可以从私钥计算得出.当用户第一次启动xdag程序时，要求用户提供一串随机字符.这些字符会作为随机生成器的种子.这些字符存在dnet_key.dat文件中.每次xdaga启动时，这些字符会参与随机数生成, 所有的私钥用这些字符生成。
+
+## 5.传输层
+
+Dagger节点利用传输协议进行区块的交换.区块的前8个字节会用做传输协议特定的信息. 需要提到的是，这个字段在计算区块散列的时候要清零.xdag程序默认使用dnet作为网络存储层.传输过程中的区块信息是加密的.使用作者的半对称算法和临时密钥对每个传输的块进行编码，并使用相同的密钥对另一侧进行解码.临时密钥由发送方生成，然后使用存储在文件dnet_key.dat中的发送方的私钥进行编码，然后使用发送方的公钥将其解码. 加密算法为长度为8192bits的RSA算法.
```
Comment (xrdavies) [inline Protocol_zh.md]: dnet is transport layer, not for storage.

**Comment by reymondtu** [comment]
ok, I move to develop and modify it

**Pull request closed without merging**